tools/wltests: fix test_series's branch names resolution

derkling opened the pull request:
When branch names are used instead of SHA1 in a series file, the name is
resolved to a full SHA1 string which is longer then the 12 chars
actually returned by the for-each-ref command we use to lookup names.

Ensure to always compare up to 12 chars of a specified SHA1.

Signed-off-by: Patrick Bellasi <patrick.bellasi@arm.com>

douglas-raillard-arm commented on tools/wltests/test_series:
```diff
@@ -812,7 +812,7 @@ name_sha1() {
 			--format='%(objectname:short) %(refname:short)' \
 			refs/heads/ refs/remotes/ refs/tags |
-	grep $COMMIT_SHA1 | awk '{print $2}' > tmp_pipe &
+	grep ${COMMIT_SHA1:0:12} | awk '{print $2}' > tmp_pipe &
```
wouldn't it be better to just use `%(refname)` in the for-each-ref command ? According to the doc, `%(refname:short)` does not give 12 chars, it gives something unambiguous which may be smaller or bigger than that.

derkling replied:
Mmm... good point, I suppose you mean for `objectname`, that's what generates the SHA1.

If I remove `:short` from objectname I get a non-ambiguous SHA1, which you right it can be longher then 12 chars. The main issues is that in a series file we don't want to use very long SHA1 either and thus, in general, we use 12 char strings which is still potentially a source of ambiguity.

Let's fix at least one ambiguity, I've updated the patch which a prefix check on the full SHA1 returned by the `for-each-ref` query...

douglas-raillard-arm replied:
Rather than prefix check, you can use `COMMIT_SHA1=$(git rev-parse --quiet --verify "$COMMIT_SHA1")` to convert the short format used in the configuration into a full-length sha1. It then becomes comparable with the full sha1 given by `%(objectname)`. If the sha1 is ambiguous, `git rev-parse` will exit with non-zero code and give an empty string so it's quite easy to catch.

douglas-raillard-arm replied:
Also I'm not 100% sure of what the code is doing but it seems like `git name-rev --name-only $COMMIT_SHA1` could replace this `for-each-ref` + `grep` + `awk`.

derkling replied:
Right... the branch name resolution code is quite convoluted right now.
Let's fix it and add in that check for unique and full SHA1s...

derkling commented:
@qperret did you have a chance to give this a try?

derkling commented:
@douglas-raillard-arm looks like the last version addressed all the comments of our previous discussion. We now do a full SHA1 check as you suggested.

AFAIK, that's ok to merge even just because it's an improvement wrt the broken version we have now. For further updates, we will always have time. ;)

douglas-raillard-arm commented:
@derkling Can you add a TODO in the code mentioning `git name-rev --name-only $COMMIT_SHA1` for a possible simplification so this does not get lost ?

derkling commented:
@douglas-raillard-arm DONE ;)
